tls/s2n_ktls_io.c

```diff
@@ -174,3 +175,38 @@ S2N_RESULT s2n_ktls_get_control_data(struct msghdr *msg, int cmsg_type, uint8_t
 
     return S2N_RESULT_OK;
 }
+
+S2N_RESULT s2n_ktls_sendmsg(struct s2n_connection *conn, uint8_t record_type, const struct iovec *msg_iov,
+        size_t msg_iovlen, s2n_blocked_status *blocked, size_t *bytes_written)
+{
+    RESULT_ENSURE_REF(bytes_written);
+    RESULT_ENSURE_REF(msg_iov);
+    RESULT_ENSURE_REF(blocked);
+    RESULT_ENSURE_REF(conn);
+
+    *blocked = S2N_BLOCKED_ON_WRITE;
+
+    struct msghdr msg = {
+        /* msghdr requires a non-const iovec. This is safe because s2n-tls does
+         * not modify msg_iov after this point.
+         */
+        .msg_iov = (struct iovec *) (uintptr_t) msg_iov,
```

Comment:
Is there a reason why msg_iov is const in the first place?

Reply:
Yep thats what we get from our public APIs: https://github.com/aws/s2n-tls/blob/main/tls/s2n_send.c#L227-L242

It makes sense. We promise to the application that we wont modify their buffer, only read from it to send to the socket.